.github/workflows/ci-workflow.yml

```diff
@@ -31,112 +34,151 @@ jobs:
+    # Trying to run this on standard runners is just going to fail due to
+    # lack of CPU resources.
+    if: ${{ vars.NR_RUNNER != '' }}
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 20.x]
-
+        node: [ 16, 18, 20 ]
+    runs-on: ${{ vars.NR_RUNNER }}
+    name: Linux / Node ${{ matrix.node }} arm64
+    timeout-minutes: 15
     steps:
-    - uses: actions/checkout@v3
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v3
-      with:
-        node-version: ${{ matrix.node-version }}
-    - name: Install Dependencies
-      run: npm ci
-    - name: Rebuild
-      run: npm run rebuild
-    - name: Run Unit
-      run: npm run unit
-    - name: Post Unit Test Coverage
-      uses: codecov/codecov-action@v3
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        directory: ./coverage/unit/
-        files: lcov.info
-        flags: unit-tests-${{ matrix.node-version }}-windows-latest
-    - name: Run Integration
-      run: npm run integration
-    - name: Post Integration Test Coverage
-      uses: codecov/codecov-action@v3
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        directory: ./coverage/integration/
-        files: lcov.info
-        flags: integration-tests-${{ matrix.node-version }}-windows-latest
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Compute cache key
+        run: echo -e "CACHE_KEY=$(shasum -a 256 package.json | cut -f1 -d ' ')" >> "$GITHUB_ENV"
+      - name: Restore modules cache
+        id: cache_restore
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/node_modules
+          key: linux-arm-node-${{ matrix.node }}-${{ env.CACHE_KEY }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: arm64
+      - uses: docker/setup-buildx-action@v3
+      - uses: docker/build-push-action@v5
+        with:
+          context: .
+          build-args: |
+            NODE_VERSION=${{ matrix.node }}
+          file: linux_arm.dockerfile
+          tags: linux_arm:node-${{ matrix.node }}
+          load: true
+          push: false
+          platforms: linux/arm64
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Run test
+        uses: addnab/docker-run-action@v3
+        with:
+          image: linux_arm:node-${{ matrix.node }}
+          options: --platform linux/arm64 -v ${{ github.workspace }}:/host
+          run: |
+            cp -R /host/node_modules . 2>/dev/null
+            rm -rf /host/node_modules 2>/dev/null
+            # npm install will fail on Node 18 every time unless we use this
+            # very odd fix:
+            # https://github.com/npm/cli/issues/4652#issuecomment-1126672629
+            npm install --verbose --maxsockets 1
+            cp -R node_modules /host/
+            npm run unit
+            # Skipping integration until we can get native arm64 runners
```

Comment:
do we have a ticket for this?

Reply:
Not yet. We can file one once we merge this.